Added framework to facilitate test-case writing (#42)
Capstone-Skynet/Capstone-Skynet.github.io: docs/validation/validation_specification.tex

\item \textbf{and Legal Personnel:} to review the validation scheme if necessary. Such personnel may include personnel from the Department of Electrical and Computer Engineering, Industry Canada, and Transport Canada
\end{itemize}

\subsection{Reading Guide}
This document organizes tests based on their component sub-systems, and whether the tests are automated or not (see Section \ref{auto} for a detailed description). Each section has a table which details the test, including a unique identifier, the requirement which it is testing, the test's status, and the last time the test was executed.

\section{Validation Overview}

\subsection{Automated and Manual Tests}\label{auto}
The validation tests employed (and this document) are split into two categories, \textit{automated tests} and \textit{manual tests}:

\textbf{Automated tests} are tests focused on tasks which are performed automatically and regularly by the device and \textit{can be monitored by the device itself}. Examples of such scenarios include self-testing systems on boot, monitoring communication status, machine learning accuracy logging, et cetera.

> \textbf{Manual tests} are tests which \textit{cannot be performed by the device itself} and must therefore be manually performed by a quality controller. These tests are performed less frequently -- some may be executed prior to every flight, while others may only be performed once. 

Reviewer: ```suggestion
\textbf{Manual tests} are tests which \textit{cannot be performed by the device itself} and must be manually performed by a quality controller. These tests are performed less frequently --- some may be executed before every flight, while others may only be performed once. 
```
